catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/query/monitor/impl/WorkspaceServiceImpl.java

```diff
@@ -163,18 +198,30 @@ public String toString() {
    *
    * @return query request
    */
-  private QueryRequest createQueryRequestForAllWorkspaceMetacards() {
-    return createQueryRequest(filterService.buildWorkspaceTagFilter());
+  private Optional<QueryRequest> createQueryRequestForSubscribedToWorkspaces() {
+    return findSubscriptions().map(this::getIdsFromSubscriptions).map(this::getQueryRequestFromIds);
   }
 
-  /**
-   * Create a query request for a list of filters where the filters are OR'ed together.
-   *
-   * @param filters filter list
-   * @return query request
-   */
-  private QueryRequest createQueryRequest(List<Filter> filters) {
-    return createQueryRequest(filterBuilder.anyOf(filters));
+  private QueryRequest getQueryRequestFromIds(Set<String> ids) {
+    return createQueryRequest(workspaceQueryBuilder.createFilter(ids));
+  }
+
+  private Set<String> getIdsFromSubscriptions(List<Map<String, Object>> subscriptions) {
+    return subscriptions
+        .stream()
+        .map(subscription -> subscription.get(ID_FIELD))
+        .filter(Objects::nonNull)
+        .map(Object::toString)
+        .collect(Collectors.toSet());
+  }
+
+  private Optional<List<Map<String, Object>>> findSubscriptions() {
```

Comment:
✏️ I think it's odd to use `Optional` with collections since you can return an empty collection instead of an empty `Optional`.

Reply:
Changed to 
```java
  private Optional<QueryRequest> createQueryRequestForSubscribedToWorkspaces() {
    final List<Map<String, Object>> subscriptions = findSubscriptions();
    if (subscriptions.isEmpty()) {
      return Optional.empty();
    }
    final Set<String> ids = getIdsFromSubscriptions(subscriptions);
    return Optional.of(getQueryRequestFromIds(ids));
  }
```

at your suggestion.